### ISSUE TITLE
some local vars of a for-statement are invisible in 2.12.12

### ISSUE THREAD

**Issue by unkarjedy** [scala/bug#12095]
## reproduction steps
Consider the example:
```scala
object InForStmt3 {
  def main(args: Array[String]) {
    for {
      a <- Seq(1)
      a1 = a + 1
    } {
      def getA = a
      def getA1 = a1
      println("")
    }
  }
}
```

compile it with scala 2.12.11 vs 2.12.12 (**same for 2.13.1 vs 2.13.3**)
After looking into the byte code you can see the following:
*scala 2.12.11:*
```
  // access flags 0x1019
  public final static synthetic $anonfun$main$2(Lscala/Tuple2;)V
    // parameter final  x$1
    ...
   FRAME FULL [scala/Tuple2 scala/runtime/BoxedUnit scala/Tuple2 I I] []
    RETURN
   L8
    LOCALVARIABLE a I L2 L1 3
    LOCALVARIABLE a1 I L4 L1 4
    LOCALVARIABLE x$1 Lscala/Tuple2; L0 L8 0
    MAXSTACK = 3
    MAXLOCALS = 5
```

*scala 2.12.12:*
```
  // access flags 0x1019
  public final static synthetic $anonfun$main$2(Lscala/Tuple2;)V
    // parameter final  x$1
    ...
   FRAME FULL [scala/Tuple2 scala/runtime/BoxedUnit scala/Tuple2 I I] []
    RETURN
   L6
    LOCALVARIABLE x$1 Lscala/Tuple2; L0 L6 0
    MAXSTACK = 3
    MAXLOCALS = 5
```

## problem
Note that scala 2.12.12 doesn't provide local variables for a & a1 anymore.
This leads to https://youtrack.jetbrains.com/issue/SCL-17927
A bigger example: https://youtrack.jetbrains.com/issue/SCL-17927#focus=Comments-27-4293827.0-0

@retronym Can be related to https://github.com/scala/scala/pull/8897

**Comment by retronym** [comment]
Fix in progress: https://github.com/retronym/scala/tree/t12095

**Comment by unkarjedy** [comment]
Sorry for spamming.
I will move this note into a separate comment just in case it was not noticed:

**The issue is also actual for **Scala 2.13.3****

**Comment by retronym** [comment]
Thanks for confirming. All fixes that target 2.12.x will be routinely forward ported to 2.13.x, so we'll get both versions.
